go/interpreter/lfvm/hash_cache.go

```diff
@@ -124,8 +124,6 @@ func (h *HashCache) getHash32(c *context, data []byte) tosca.Hash {
 			entry.pred.succ = entry.succ
 			if entry.succ != nil {
 				entry.succ.pred = entry.pred
-			} else {
-				h.tail32 = entry.pred
```

Comment:
This line of code is triggered if the very last element in the cache is touched and moved to the front. An example test provoking it is this:
```
func TestHashCache_TouchLastElement(t *testing.T) {
	ctxt := &context{}
	cache := newHashCache(3, 3)
	cache.getHash32(ctxt, []byte{byte(1)})
	cache.getHash32(ctxt, []byte{byte(2)})
	cache.getHash32(ctxt, []byte{byte(3)})
	cache.getHash32(ctxt, []byte{byte(1)})
}
```
Without this line the tail would end up pointing to the head when touching the last element in the LRU list and the LRU order is destroyed.